src/lib.rs

```diff
@@ -209,6 +210,44 @@ pub trait Device: AsRawFd {
 
         Ok(driver)
     }
+
+    /// Waits for a vblank.
+    fn wait_vblank(
+        &self,
+        target_sequence: VblankWaitTarget,
+        flags: VblankWaitFlags,
+        high_crtc: u32,
+        user_data: usize,
+    ) -> Result<VblankWaitReply, SystemError> {
+        let high_crtc_shift = drm_ffi::_DRM_VBLANK_HIGH_CRTC_SHIFT;
+        let high_crtc_mask =
+            drm_ffi::drm_vblank_seq_type::_DRM_VBLANK_HIGH_CRTC_MASK >> high_crtc_shift;
+        if (high_crtc & !high_crtc_mask) != 0 {
+            return Err(SystemError::InvalidArgument);
+        }
```

Comment:
Nit: could you save on some code/lines by pre-shifting `high_crtc << drm_ffi::_DRM_VBLANK_HIGH_CRTC_SHIFT` and:

1. Using that result below directly;
2. Use the mask against it directly, instead of also shifting the mask (which should be a constant operation, I hope the compiler is able to understand that for `high_crtc_shift` being a `let` binding instead of a `const` binding, which I don't think is needed for readability FWIW).

Reply:
That's a real outlier, but fair enough :smile:

Reply:
In any case, not sure if the `let high_crtc_shift` really helps with readability compared to a `const` or even importing it in scope.

Reply:
Do you mean something like this?
```rust
fn wait_vblank(...) -> ... {
    use drm_ffi::drm_vblank_seq_type::_DRM_VBLANK_HIGH_CRTC_MASK as HIGH_CRTC_MASK;
    use drm_ffi::_DRM_VBLANK_HIGH_CRTC_SHIFT as HIGH_CRTC_SHIFT;
    
    let high_crtc_mask = HIGH_CRTC_MASK >> HIGH_CRTC_SHIFT;
    if (high_crtc & !high_crtc_mask) != 0 {
        return Err(SystemError::InvalidArgument);
    }
```

Reply:
Yeah, that way they stay const without having to do `const THE_SHORTER_NAME: the_type = drm_ffi::..;`, though not sure if you need the `as` rename anyway but could just use the full name without the `drm_ffi::` in the rest of the code.

Reply:
Fair point. Changes pushed.